src/cctag/Params.cpp

```diff
@@ -79,6 +81,11 @@ Parameters::Parameters(std::size_t nCrowns)
     }
 }
 
+Parameters::~Parameters( )
```

Comment:
Does `Parameters::~Parameters() = default;` work as well (in terms of the double-free error you mentioned)? (it would be better than an empty destructor with just a debugging cout)

Reply:
I have tried with the default destructor and it does not work. It is required to have an empty constructor to avoid the race in deletion.